crates/db_schema/src/schema_setup/diff_check.rs

```diff
@@ -0,0 +1,291 @@
+use lemmy_utils::settings::SETTINGS;
+use std::{
+  borrow::Cow,
+  collections::btree_set::{self, BTreeSet},
+  process::{Command, Stdio},
+};
+
+// It's not possible to call `export_snapshot()` for each dump and run the dumps in parallel with
+// the `--snapshot` flag. Don't waste your time!!!!
+
+pub fn get_dump() -> String {
+  let db_url = SETTINGS.get_database_url();
+  let output = Command::new("pg_dump")
+    .args([
+      // Specify database URL
+      "--dbname",
+      &db_url,
+      // Allow differences in row data and old fast tables
+      "--schema-only",
+      "--exclude-table=comment_aggregates_fast",
+      "--exclude-table=community_aggregates_fast",
+      "--exclude-table=post_aggregates_fast",
+      "--exclude-table=user_fast",
+      // Ignore some things to reduce the amount of queries done by pg_dump
+      "--no-owner",
+      "--no-privileges",
+      "--no-comments",
+      "--no-publications",
+      "--no-security-labels",
+      "--no-subscriptions",
+      "--no-table-access-method",
+      "--no-tablespaces",
+    ])
+    .stderr(Stdio::inherit())
+    .output()
+    .expect("failed to start pg_dump process");
+
+  // TODO: use exit_ok method when it's stable
+  assert!(output.status.success());
+
+  String::from_utf8(output.stdout).expect("pg_dump output is not valid UTF-8 text")
+}
+
+const PATTERN_LEN: usize = 19;
+
+pub fn check_dump_diff(before: String, after: String, label: &str) {
```

Comment:
Why is so much custom code necessary for checking a diff?

Reply:
The added complexity is now in #5204 instead of this PR, and I will try to make the comments more clear about the reason in that PR